app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -134,8 +134,10 @@ class AgendaViewModel @Inject constructor(
                 }
             }
             is ToggleTaskCompleted -> {
-                _agendaState.value.agendaItems.mapLatest {
-                    it.find { item -> item.id == event.agendaItemId }?.let { task ->
+                _agendaState.value.agendaItems.collectLatest {
+                    it.find {
+                        item -> item.id == event.agendaItemId
+                    }?.let { task ->
```

Comment:
Why do you use a collect block in here? If this gets called, the collector will block this function until it stopped collecting, I don't think this is what you want 🤔 
I'd either pass the Task that was clicked on from the UI or simply use `_agendaState.value.agendaItems.find {}`

Reply:
OK, i will refactor this.